### ISSUE TITLE
xMSannotator (simple) fails unpredictably

### ISSUE THREAD

**Issue by martenson** [RECETOX/galaxytools#16]
example history: https://umsa.cerit-sc.cz/u/6b58dfad0690401d867862bccc591cb7/h/unnamed-history

```
Error in H5Fopen(file, native = native) : HDF5. File accessibilty.Unable to open file. Calls: <Anonymous> -> h5write.default -> h5checktypeOrOpenLoc -> H5Fopen
```

~~tool works if datasets are uploaded directly~~ not true, error is intermittent

**Comment by ondrejme** [comment]
Similar problem here [https://github.com/MaayanLab/archs4/issues/6](https://github.com/MaayanLab/archs4/issues/6), apparently solved by downgrading some package. Seems like a R/bioconda issue rather than galaxy having trouble accessing Sally files, but I might be wrong.

**Comment by ondrejme** [comment]
Might be related to #14

**Comment by xtracko** [comment]
> 
> 
> Similar problem here [MaayanLab/archs4#6](https://github.com/MaayanLab/archs4/issues/6), apparently solved by downgrading some package. Seems like a R/bioconda issue rather than galaxy having trouble accessing Sally files, but I might be wrong.

I made 2 runs from the same docker image on Galaxy but on differently stored data (the content of the files was same but it only differed that in the second run it was picked throug data library rather than local storage) and it succeded in one. Just to be sure, a made one additional run on my local PC of the docker image and it also succeded. Therefore, it is not R related because the docker environment is isolated and consistent.

Also, look at the shared Galaxy history regarding aplcms. Last week it failed with the same message (cant open file for writing), yesterday it finished succesfuly. I haven't changed anything in the wraper nor in the docker image. But something must have hapend with Galaxy.

**Comment by ondrejme** [comment]
Docker environment might be consistent, but probably not stable. If you read through the issue I posted, they mention that downloading the file from somewhere remote sometimes doesn't work. See image

![image](https://user-images.githubusercontent.com/53870015/94808902-21dc8080-03f2-11eb-81b7-3f19daa4f884.png)

**Comment by martenson** [comment]
Some more possibly related conversations: https://github.com/pachterlab/sleuth/issues/120

**Comment by martenson** [comment]
This could be related: https://msmith.de/2017/06/23/r-download-file.html

Especially since our "peaks" datasets reports different size on each upload (ping @ondrejme)

**Comment by xtracko** [comment]
I don't suspect that this is caused by the linked issues (no downloading of files is happening and the docker image works in 3 different environments of mine, including native windows) but nevertheless something fishy is going on underneath. I will try to use a different library for H5 files. I spoke to @smartx-usman and he said that Retip is using the hdf5r library (I am using rhdf5). Let us see whether it fixes the problem.

**Comment by xtracko** [comment]
Fixed in #22
